r1.sh

```diff
@@ -65,20 +75,38 @@ python3 -m venv venv
 source venv/bin/activate
 
 chmod +x mtkbootcmd.py
+# Make `mtkbootcmd.py` use "${ESCAPED_PWD}"/venv/bin/python.
+if [[ "$OSTYPE" == "darwin"* ]]; then
+    # Sed is different on mac.
+    grep -rl "usr/bin/env python3" mtkbootcmd.py | xargs sed -i '' $SEDSTR_ENVPYTHON
+elif [[ "$OSTYPE" == "linux-gnu"* ]]; then
+    grep -rl "usr/bin/env python" mtkbootcmd.py | xargs sed -i $SEDSTR_ENVPYTHON
+fi
 
 # Download mtkclient
 REPO_URL="https://github.com/AgentFabulous/mtkclient"
 REPO_NAME=$(basename "$REPO_URL" .git)
 git clone "$REPO_URL"
 cd "$REPO_NAME" || exit
-pip3 install -r requirements.txt
+
+# Make mtkclient use "${ESCAPED_PWD}"/venv/bin/python, for consistency.
+if [[ "$OSTYPE" == "darwin"* ]]; then
+    # Sed is different on mac.
+    grep -rl "usr/bin/python3" . | xargs sed -i '' $SEDSTR_BINPYTHON
+    grep -rl "usr/bin/env python3" . | xargs sed -i '' $SEDSTR_ENVPYTHON
+elif [[ "$OSTYPE" == "linux-gnu"* ]]; then
+    grep -rl "usr/bin/python" . | xargs sed -i $SEDSTR_BINPYTHON
+    grep -rl "usr/bin/env python" . | xargs sed -i $SEDSTR_ENVPYTHON
```

Comment:
Can't this be a change directly in mtkclient? (To use `usr/bin/env` instead of `usr/bin/python(3)`)

Reply:
No, because `/usr/bin/env` may not point to the virtualenv created in the earlier step depending on how it's being run. This is hard coding it to use the virtualenv.